homeassistant/components/lamarzocco/calendar.py

```diff
@@ -0,0 +1,101 @@
+"""Calendar platform for La Marzocco espresso machines."""
+from collections.abc import Iterator
+from datetime import datetime, timedelta
+
+from homeassistant.components.calendar import CalendarEntity, CalendarEvent
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util import dt as dt_util
+
+from .const import DOMAIN
+from .entity import LaMarzoccoBaseEntity
+
+CALENDAR_KEY = "auto_on_off_schedule"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up switch entities and services."""
+
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities([LaMarzoccoCalendarEntity(coordinator, CALENDAR_KEY)])
+
+
+class LaMarzoccoCalendarEntity(LaMarzoccoBaseEntity, CalendarEntity):
+    """Class representing a La Marzocco calendar."""
+
+    _attr_translation_key = CALENDAR_KEY
+
+    @property
+    def event(self) -> CalendarEvent | None:
+        """Return the next upcoming event."""
+        # only need to check the next 6 days, because if we don't find anything there
+        # then there is no event scheduled
+        for date in self._get_date_range(
+            dt_util.now(), dt_util.now() + timedelta(days=6)
+        ):
+            scheduled = self._async_get_calendar_event(date)
+            if scheduled:
+                if scheduled.start < dt_util.now() and scheduled.end < dt_util.now():
```

Comment:
Lets extract a `now` variable at the top of this function and use in the 4 calls.